python/format/src/errors.py

```diff
@@ -0,0 +1,62 @@
+import contextlib
+import dataclasses
+from typing import List, Union
+
+
+class ValidationError(Exception):
+    """Error during the validation of the format."""
+
+
+@dataclasses.dataclass
+class Issues:
+    """
+    Issues during the validation of the format.
+
+    Issues can either errors (blocking) or warnings (informative).
+    """
+
+    errors: List[str] = dataclasses.field(default_factory=list)
+    warnings: List[str] = dataclasses.field(default_factory=list)
+    _local_context: Union[str, None] = None
+
+    def _wrap_in_local_context(self, issue: str) -> str:
+        if self._local_context is None:
+            return issue
+        return f"[{self._local_context}] {issue}"
+
+    def add_error(self, error: str):
+        """Mutates self.errors with a new error."""
+        self.errors.append(self._wrap_in_local_context(error))
+
+    def add_warning(self, warning: str):
+        """Mutates self.warnings with a new warning."""
+        self.warnings.append(self._wrap_in_local_context(warning))
+
+    def report(self) -> str:
+        """Reports errors and warnings in a string."""
+        message = ""
+        for issues, issue_type in [
+            (self.errors, "error(s)"),
+            (self.warnings, "warning(s)"),
+        ]:
+            num_issues = len(issues)
+            if num_issues:
+                message += f"\nFound the following {len(issues)} {issue_type} during the validation:"
```

Comment:
Curious why the \n is at the beginning, not at the end of the messages.

Reply:
Done.